benchmarking/benchmark.py

```diff
@@ -273,203 +80,73 @@ def save_results(
-        "Atom Pairs Fingerprint",
-        True,
-    )
-
-    # TOPOLOGICAL TORSION FINGERPRINT
-    print("Topological Torsion")
-    topological_torsion_skfp_times = get_all_times_skfp(
-        X, TopologicalTorsionFingerprint
-    )
-    generator = fpgens.GetTopologicalTorsionGenerator()
-    topological_torsion_sequential_times = get_all_generator_times_rdkit(X, generator)
-    save_all_results(
-        topological_torsion_skfp_times,
-        topological_torsion_sequential_times,
-        n_molecules,
-        "Topological Torsion Fingerprint",
-        True,
-    )
-
-    # MACCS KEYS FINGERPRINT
-    print("MACCS Keys")
-    MACCSKeys_skfp_times = get_all_times_skfp(X, MACCSKeysFingerprint, False)
-    MACCSKeys_sequential_times = get_all_sequential_times(
-        X, GetMACCSKeysFingerprint, False
-    )
-    save_all_results(
-        MACCSKeys_skfp_times,
-        MACCSKeys_sequential_times,
-        n_molecules,
-        "MACCS Keys fingerprint",
-        False,
-    )
-
-    # ERG FINGERPRINT
-    print("ErG")
-    ERG_skfp_times = get_all_times_skfp(X, ERGFingerprint, False)
-    ERG_sequential_times = get_all_sequential_times(X, GetErGFingerprint, False)
-    save_all_results(
-        ERG_skfp_times,
-        ERG_sequential_times,
-        n_molecules,
-        "ErG fingerprint",
-        False,
-    )
-
-    # MAP4 FINGERPRINT
-    print("MAP4")
-    MAP4_skfp_times = get_all_times_skfp(X, MAP4Fingerprint)
-    MAP4_sequential_times = get_all_sequential_times(X, get_map4_fingerprint)
-    save_all_results(
-        MAP4_skfp_times,
-        MAP4_sequential_times,
-        n_molecules,
-        "MAP4 fingerprint",
-        True,
-    )
-
-    # MHFP FINGERPRINT
-    print("MHFP")
-    MHFP_skfp_times = get_all_times_skfp(X, MHFPFingerprint)
-    MHFP_sequential_times = get_all_sequential_times(X, get_mhfp)
-    save_all_results(MHFP_skfp_times, MHFP_sequential_times, n_molecules, "MHFP", True)
-
-    # Avalon FINGERPRINT
-    print("Avalon")
-    Avalon_skfp_times_count = get_all_times_skfp(X, AvalonFingerprint, True)
-    Avalon_sequential_times = [
-        get_all_sequential_times(X, GetAvalonFP, False),
-        get_all_sequential_times(X, GetAvalonCountFP, False),
+    print(f"Number of molecules : {n_molecules}")
+
+    fingerprint_constructors = [
+        AtomPairFingerprint,
+        AutocorrFingerprint,
+        AvalonFingerprint,
+        # E3FPFingerprint,
+        ECFPFingerprint,
+        ERGFingerprint,
+        EStateFingerprint,
+        # GETAWAYFingerprint,
+        LayeredFingerprint,
+        MACCSFingerprint,
+        MAPFingerprint,
+        MHFPFingerprint,
+        # MordredFingerprint,
+        # MORSEFingerprint,
+        PatternFingerprint,
+        # PharmacophoreFingerprint,
+        PhysiochemicalPropertiesFingerprint,
+        PubChemFingerprint,
+        # RDFFingerprint,
+        RDKitFingerprint,
+        SECFPFingerprint,
+        TopologicalTorsionFingerprint,
+        # WHIMFingerprint,
```

Comment:
Uncomment 3D fingerprints, just compute and pass conformers using `ConformerGenerator`